### PR TITLE
fix(ui): show popup variation rule message while hovering on variation delete button

### PR DESCRIPTION
**Pull request by bimalgrg519**
Fix https://github.com/bucketeer-io/bucketeer/issues/142 https://github.com/bucketeer-io/bucketeer/issues/138

### CODE REVIEW

**Comment by cre8ivejp** [comment]
Thank you for your PR!

I deployed it and found a bug when the same variation is being used in the default and off variation. It shows the HoverPopover without a message.

Screen Shot 2022-12-31 at 19 57 53

It will be easier to understand and simple if we don't show the delete button when there are only two variations. What do you think?
Also, can you limit the popover's width to 350px?

**Comment by bimalgrg519** [comment]
When there are only two variations and lets say no rule being applied on second variation then we can hide delete button on second variation right?

**Comment by cre8ivejp** [comment]
@bimalgrg519 

> When there are only two variations and lets say no rule being applied on second variation then we can hide delete button on second variation right?

No. If we have only two variations, there is no need to check if there is a rule to the second variation.
We check the list length, and if it is two, we hide the delete button.
If we have three or more variations, then we show the delete buttons using the conditions that you implemented in the PR.

**Comment by bimalgrg519** [comment]
Thank you for the explanation. Sent PR.

**Comment by cre8ivejp** [comment]
@bimalgrg519 

I found that when you have four variations and delete variation 1, it doesn't update the delete button status. Can you take a look, please?
http://localhost:8000/development/features/feature-flag-string/variation

Added four variations
Screen Shot 2023-01-01 at 19 03 59

After you removed variation 1.
Screen Shot 2023-01-01 at 19 04 11

**Review by cre8ivejp — CHANGES_REQUESTED** [review]

File: ui/web-v2/apps/admin/src/components/VariationInput/index.tsx [hunk 1]
```diff
@@ -162,9 +162,9 @@ export const VariationInput: FC<VariationInputProps> = memo(
           case typeDisabled && variationId === offVariationId:
             return f(messages.feature.variationSettings.offVariation);
 
-          // Check if the variation is both the on variation and the off variation
+          // Check if the variation is both the on and off variation
```
Comment (cre8ivejp) [inline ui/web-v2/apps/admin/src/components/VariationInput/index.tsx:165]: nit: `Check if the variation is both on and off variations`

File: ui/web-v2/apps/admin/src/components/VariationInput/index.tsx [hunk 2]
```diff
@@ -148,11 +148,11 @@ export const VariationInput: FC<VariationInputProps> = memo(
       (variationId) => {
         // Use a switch statement to determine the message
         switch (true) {
-          // Check if the variation is both the on variation and the off variation
+          // Check if the variation is both the on and off variations
```
Comment (cre8ivejp) [inline ui/web-v2/apps/admin/src/components/VariationInput/index.tsx:151]: nit: `Check if the variation is both on and off variations`

File: ui/web-v2/apps/admin/src/components/VariationInput/index.tsx [hunk 3]
```diff
@@ -162,7 +162,7 @@ export const VariationInput: FC<VariationInputProps> = memo(
           case typeDisabled && variationId === offVariationId:
             return f(messages.feature.variationSettings.offVariation);
 
-          // Check if the variation is both the on and off variation
+          // Check if the variation is both on and off variation
```
Comment (cre8ivejp) [inline ui/web-v2/apps/admin/src/components/VariationInput/index.tsx:165]: nit: `Check if the variation is both on and off variations`

**Review by cre8ivejp — APPROVED** [review]
Thank you!!!

**Pull request merged**